src/base/http/server.cpp

```diff
@@ -108,9 +109,13 @@ void Server::incomingConnection(const qintptr socketDescriptor)
         static_cast<QSslSocket *>(serverSocket)->startServerEncryption();
     }
 
-    auto *c = new Connection(serverSocket, m_requestHandler, this);
-    m_connections.insert(c);
-    connect(serverSocket, &QAbstractSocket::disconnected, this, [c, this]() { removeConnection(c); });
+    Connection *socket = new Connection(serverSocket, this);
+    m_connections.insert(socket);
+    connect(socket, &Connection::disconnected, this, [socket, this]() { removeConnection(socket); });
+    connect(socket, &Connection::readyRequest, this, [socket, this]()
+    {
+        m_connectionHandler->handleConnection(socket);
```

Comment:
Seems it handles not Connection itself but each request received within Connection.

Reply:
hmm, then I will have to use `QObject::Connect` in non QObject classes, I don't think you would like it either.